### PR TITLE
Support deregister field in Check.script

### PR DESCRIPTION
**Pull request by mougams**

### CODE REVIEW

**Review by pierrecdn — CHANGES_REQUESTED** [review]
I think you've pushed way to much commits here. Missing a rebase on upstream?

File: consul/base.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ class Check:
     There are three different kinds of checks: script, http and ttl
     """
     @classmethod
-    def script(klass, args, interval):
+    def script(klass, args, interval, deregister=None):
```
Comment (pierrecdn) [inline consul/base.py:51]: Shouldn't we keep the exact same semantic, ie. name the argument `deregister_critical_service_after`?

Comment (pierrecdn) [reply consul/base.py:51]: Hmm, OK you can forget this request, since it's not this way for other checks (http, tcp).

**Comment by mougams** [comment]
> I think you've pushed way to much commits here. Missing a rebase on upstream?

It is now fixed

**Review by pierrecdn — APPROVED** [review]

**Pull request merged**